ui/app/mainui/Popups.qml

```diff
@@ -461,6 +467,32 @@ QtObject {
             DiscordImportProgressDialog {
                 store: root.communitiesStore
             }
+        },
+
+        Component {
+            id: deleteMessageConfirmationDialogComponent
+            DeleteMessageConfirmationPopup {
+                onClosed: destroy()
+            }
+        },
+
+        Component {
+            id: downloadImageDialogComponent
+            FileDialog {
```

Comment:
And since we're already using the "platform" components, what about https://doc.qt.io/qt-5/qml-qt-labs-platform-folderdialog.html? This works much better, doesn't depend on QtQuick.Controls1 and it also has the `closed()` signal

Reply:
Ok, not really a big issue since we are using them elsewhere; I just felt this refactor is a good opportunity to do so